db/change_cache.go

```diff
@@ -277,8 +277,9 @@ func (c *changeCache) InsertPendingEntries(ctx context.Context) error {
 	// Trigger _addPendingLogs to process any entries that have been pending too long:
 	c.lock.Lock()
 	changedChannels := c._addPendingLogs(ctx)
-	if c.notifyChange != nil && len(changedChannels) > 0 {
-		c.notifyChange(ctx, changedChannels)
+	if c.notifyChangeFunc != nil && len(changedChannels) > 0 {
+		channelSet := channels.SetFromArrayNoValidate(changedChannels)
+		c.notifyChangeFunc(ctx, channelSet)
 	}
```

Comment:
I don't think we want to lock here.

Reply:
That's strange we seem to hold the lock for notify here, will move the unlock to after addPendingLogs